components/script/script_thread.rs

```diff
@@ -574,6 +576,17 @@ impl ScriptThreadFactory for ScriptThread {
 }
 
 impl ScriptThread {
+    pub fn note_newly_transitioning_nodes(nodes: Vec<UntrustedNodeAddress>) {
```

Comment:
Can GC happen before this even gets a change to be called?

IMO this whole function should be unsafe.